src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/PathConverter.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
+using System.Collections.Generic;
+using System.Linq;
+using ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using System.Collections.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
+using System.IO;
+
+internal class PathConverter : IPathConverter
+{
+    // The path on this computer to which we deployed the test dll and test runner
+    private readonly string _deploymentPath = "";
+    // The path on the remote system where test dll was originally placed, and from which we
+    // copied it to this system. For vstest.console, which is on the other side of this, the names
+    // are inverted, it sends us their local path, and thinks about our local path as remote.
+    private readonly string _originalPath = "";
+
+    public PathConverter(string originalPath, string deploymentPath, IFileHelper fileHelper)
```

Comment:
```suggestion
    public PathConverter(string originalPath!!, string deploymentPath!!, IFileHelper fileHelper!!)
```

Reply:
doing this will complain about formatting in vs 🤷‍♀️ was the other PR not merged yet? I also check that the values are there otherwise I use the null path converter, so this should be reasonably safe.

Reply:
You need to install 6.0.3 preview to be able to run inside VS

Reply:
I forgot to update main obviously.